### PR TITLE
Improve Error Message for throwSqlError()

### PR DESCRIPTION
**Pull request by DanRod1999**
imodel-native: https://github.com/iTwin/imodel-native/pull/905
Resolves [Issue](https://github.com/iTwin/itwinjs-core/issues/7299)

Adds test for new sql error message. See imodel-native pr for changes

### CODE REVIEW

File: core/backend/src/test/imodel/IModel.test.ts [hunk 1]
```diff
@@ -2844,4 +2844,42 @@ describe("iModel", () => {
 
     expect(() => imodel.elements.getElement<Subject>(IModel.rootSubjectId)).to.throw(IModelError, "Element=0x1", "Not Found");
   });
+
+  it("should throw \"Error: constraint failed (BE_SQLITE_CONSTRAINT_UNIQUE)\" to insert instance with same props twice", () => {
```
Comment (nick4598) [inline core/backend/src/test/imodel/IModel.test.ts:2848]:
```suggestion
  it("should throw \"Error: constraint failed (BE_SQLITE_CONSTRAINT_UNIQUE)\" when inserting instance with same props twice", () => {
```

File: core/backend/src/test/imodel/IModel.test.ts [hunk 2]
```diff
@@ -2844,4 +2844,42 @@ describe("iModel", () => {
...
+      category: spatialCategoryId,
+      code: Code.createEmpty(),
+    };
+
+    const id0 = elements.insertElement(elementProps);
+    const id1 = elements.insertElement(elementProps);
+
+    // // Create grouping relationships from 0 to 1 and from 0 to 2
+    // const r1 = ElementGroupsMembers.create(imodel, id0, id1, 1);
+    // r1.insert();
```
Comment (nick4598) [inline core/backend/src/test/imodel/IModel.test.ts:2872]:
```suggestion
```

**Review by nick4598 — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

**Review by tcobbs-bentley — APPROVED** [review]
Mobile version updates look fine.

**Comment by aruniverse** [comment]
@mergifyio backport release/4.10.x release/4.11.x

**Pull request merged**